components/square.tsx

```diff
@@ -1,14 +1,30 @@
+import {Button, Skeleton} from "@mui/material";
+import {Close, RadioButtonUnchecked} from "@mui/icons-material";
+
 interface SquareProps {
     onClick: () => void;
     value: any;
 }
 
 export default function Square(props: SquareProps) {
+
+    function getIcon() {
+        if (!props.value) {
+            return <Skeleton variant={"text"} width={16} animation={false} />;
+        }
+        return props.value === 'X' ?
+            <Close />
+            : <RadioButtonUnchecked />
+    }
+
+
     return (
-        <button
-            className={"square"}
-            onClick={() => props.onClick()}>
-            {props.value}
-        </button>
+        <Button
+            disabled={props.value}
+            variant={"outlined"}
+            onClick={() => props.onClick()}
+        fullWidth={true}>
+            {getIcon()}
+        </Button>
```

Comment:
```suggestion
        <Button
            disabled={props.value}
            variant={"outlined"}
            onClick={() => props.onClick()}
        fullWidth={true}>
            {!props.value && <Skeleton variant={"text"} width={16} animation={false} />}
            {props.value === 'X' && <Close />}
            {props.value === 'O' && <RadioButtonUnchecked />}
        </Button>
```

This one is just a preference but if you wanna use a function you should combine it with useCallback.
https://reactjs.org/docs/hooks-reference.html#usecallback